README.md

```diff
@@ -1,68 +1,75 @@
+A big part of the power in these simple interface functions is that each (`Tables.rows` & `Tables.columns`) is defined for any table type, even if the table type only explicitly implements one interface function or the other.
+This is accomplished by providing performant, generic fallback definitions in Tables.jl itself (though obviously nothing prevents a table type from implementing each interface function directly).
 
-Let's clarify the definitions of some of the terms that were just used:
+With these simple definitions, powerful workflows are enabled:
+* A package providing data cleansing, manipulation, visualization, or analysis can automatically handle any number of decoupled input table types
+* A tabular file format can have automatic integration with in-memory structures and translation to other file formats
 
-* `Tables.rows(src)` returns a `Row` iterator, where `Row` is any object that supports value access via `getproperty(row, nm::Symbol)`. For example, if I have a NamedTuple like `row = (a=1, b=2, c=3)`, its values can be accessed like `row.a`, which is desugared to a call to `getproperty(row, :a)`. Thus a NamedTuple implicitly satisfies the `Row` interface. A `Row` should support `getproperty(row, name)` for any column name returned in a table's `Tables.schema(src)`. This allows an end-user access pattern like:
+So how does one go about satisfying the Tables.jl interface functions? It mainly depends on what you've alrady defined and the natural access patterns of your table:
 
-```julia
-sch = Tables.schema(src)
-names = Tables.names(sch)
-for row in Tables.rows(src)
-    for nm in names
-        foo(getproperty(row, nm))
-    end
-end
-```
-* `Tables.columns(src)` returns an object that satisifes the `Columns` interface, which, similar to a `Row`, allows access to values via `getproperty(columns, nm::Symbol)`, but, unlike a `Row` returning a scalar value, will return a vector or iterator representing all the values in the column. 
+To support `Rows`:
+* Define `Tables.rowaccess(::Type{MyTable}) = true`: this signals to other types that `MyTable` supports valid `Row`-iteration
+* Define `Tables.rows(x::MyTable)`: return a `Row`-iterator object (perhaps the table itself if already defined)
 
+To support `Columns`:
+* Define `Tables.columnaccess(::Type{MyTable}) = true`: this signals to other types that `MyTable` supports returning a valid `Columns` object
+* Define `Tables.columns(x::MyTable)`: return a `Columns`, property-accessible object (perhaps the table itself if it naturally supports property-access to columns)
 
-So how does one go about satisfying these interface functions as a table type developer? It mainly depends on the `Tables.AccessStyle(T)` of your table:
+In addition, it can be helpful to define the following:
+* `Tables.istable(::Type{MyTable}) = true`: there are runtime checks to see if an object implements the interface, but this can provide an explicit affirmation
```

Comment:
I'd go as far as making this mandatory. Currently, checking whether a type implements a method is slow AFAIK, and it's cleaner to clearly require types to implement this trait. Else nobody can really rely on it, which makes it not that useful.